test/extensions/filters/http/set_filter_state/integration_test.cc

```diff
@@ -66,6 +67,44 @@ class SetMetadataIntegrationTest : public testing::Test {
     EXPECT_EQ(Http::FilterHeadersStatus::Continue, filter->decodeHeaders(headers_, true));
   }
 
+  void runPerRouteFilter(const std::string& listener_yaml_config,
+                         const std::string& per_route_yaml_config) {
+    Server::GenericFactoryContextImpl generic_context(context_);
+
+    envoy::extensions::filters::http::set_filter_state::v3::Config listener_proto_config;
```

Comment:
nit: this actually is filter level rather than listener level, it would be better to name it filter_proto_config